packages/platforms/electron-player/src/main/api/file-system.ts

```diff
@@ -159,11 +162,19 @@ export async function deleteFile(
   storagePath: string,
   url: string
 ): Promise<void> {
-  const fullPath = join(BASE_DIR, storagePath);
+  const localPath = getLocalPath(storagePath, url);
+  await deleteFileIfExists(localPath);
+}
+
+async function deleteFileIfExists(filePath: string): Promise<void> {
   try {
-    const filePath = join(fullPath, getFileName(url));
     await unlink(filePath);
-  } catch (error) {
+  } catch (error: any) {
+    if (error?.code === 'ENOENT') {
```

Comment:
But that would bypass type safety since there is no guarantee that an error object was actually thrown. If we cast to Error but someone throws for example a string like so `throw 'This is stupid!';` or even worse `throw undefined;` we would get a runtime error if we cast to Error in the catch.